addons/base_vat/models/res_partner.py

```diff
@@ -208,7 +208,9 @@ def check_vat(self):
             # triggers two distinct write on res.partner, one for each field, both triggering this constraint.
             # If vat is set before country_id, the constraint must not break.
 
-            if failed_check:
+            # The context 'no_vat_validation' allows you to store/set a VAT number without doing validations.
+            # This is for API pushes from external platforms where you have no control over VAT numbers.
+            if failed_check and not self.env.context.get('no_vat_validation'):
```

Comment:
Rather put it at the beginning of the function, just doing a return if this context key is set. Here, you're still running the check (potentially connecting to a web service) for nothing.